pkg/controllers/rollout/controller.go

```diff
@@ -385,14 +441,26 @@ func pickBindingsToRoll(allBindings []*fleetv1beta1.ClusterResourceBinding, late
 	upperBoundReadyNumber := len(canBeReadyBindings)
 	maxNumberToAdd := maxReadyNumber - upperBoundReadyNumber
 
-	klog.V(2).InfoS("Calculated the max number of bindings to add", "clusterResourcePlacement", klog.KObj(crp),
+	klog.V(2).InfoS("Calculated the max number of bindings to add", "clusterResourcePlacement", crpKObj,
 		"maxSurgeNumber", maxSurgeNumber, "maxReadyNumber", maxReadyNumber, "upperBoundReadyBindings",
 		upperBoundReadyNumber, "maxNumberOfBindingsToAdd", maxNumberToAdd)
-	for i := 0; i < maxNumberToAdd && i < len(boundingCandidates); i++ {
-		toBeUpdatedBinding = append(toBeUpdatedBinding, boundingCandidates[i])
+
+	// boundingCandidatesUnselectedIndex stores the last index of the boundingCandidates which are not selected to be updated.
+	// The rolloutStarted condition of these elements from this index should be updated.
+	boundingCandidatesUnselectedIndex := 0
+	for ; boundingCandidatesUnselectedIndex < maxNumberToAdd && boundingCandidatesUnselectedIndex < len(boundingCandidates); boundingCandidatesUnselectedIndex++ {
```

Comment:
It' just a personal opinion, but, would it be easier if we just re-slice the array?

Reply:
add a todo there, there are few places. Could do the refactor in a separate PR.